packages/backend-data/src/convert_authorization_modes.ts

```diff
@@ -95,14 +97,38 @@ const convertOIDCAuthConfigToCDK = ({
 
 /**
  * Compute default auth mode based on availability of auth resources.
- * Will specify userPool if both user pool is specified, and no auth resources are specified, else rely on override value if needed.
+ *
+ * Will default to userPool if userPool is provided and auth resources are not provided.
+ *
+ * If no auth resources are provided and there's only one mode we'll use that as we implicitly always add iam auth.
  */
 const computeDefaultAuthorizationMode = (
   providedAuthConfig: ProvidedAuthConfig | undefined,
   authModes: AuthorizationModes | undefined
 ): DefaultAuthorizationMode | undefined => {
-  if (providedAuthConfig && !authModes) return 'userPool';
-  return;
+  if (isUsingDefaultApiKeyAuth(providedAuthConfig, authModes)) {
+    return 'apiKey';
+  } else if (providedAuthConfig && !authModes) {
+    return 'userPool';
+  } else if (
+    !providedAuthConfig &&
+    authModes &&
+    Object.keys(authModes).length === 1
+  ) {
+    if (authModes.oidcAuthorizationMode) {
```

Comment:
nit: would be slightly quicker to use `switch`

Reply:
I'll give it a shot separately. If it looks better I'll include it in separate PR. (don't want to go into feature branch PR dance now).